department-of-veterans-affairs/vets-website
# Cc/95281/review page

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

Updates the review page from `ConfirmReviewPage` to `ReviewAndConfirm` and bring the page up to the current figma design. Also adds selection of mock data and tests. Also breaks out provider address and phone into a component that can be used on the choose page and review page.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#95281

## Testing done

Unit tests

## Screenshots

![localhost_3001_my-health_appointments_schedule-referral_review_id=add2f0f4-a1ea-4dea-a504-a54ab57c6800](https://github.com/user-attachments/assets/16756588-0404-4215-a667-dc57ebf5ee34)


## What areas of the site does it impact?

Community care referral self scheduling feature in VAOS

## Acceptance criteria
 - [ ] Route to page works
 - [ ] Page uses mock data for provider and referral
 - [ ] Reload refetches the provider and gets saved slot from session storage
 - [ ] Redirects back to beginning if no selected slot in redux or session
 - [ ] back buttons route back properly
 - [ ] passes accessibility tests
 
### Quality Assurance & Testing


### Error Handling


### Testing steps
 - pull this branch locally
 - turn on the `vaOnlineSchedulingCCDirectScheduling` feature
 - run local mocks and start the app
 - navigate to http://localhost:3001/my-health/appointments/referrals-requests
 - click `Schedule your appointment` on any of the referrals
 - click `Schedule your appointment` on the next page as well
 - select a slot on the schedule page and click continue
 - confirm the review page matches the [figma](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=2217-67077&t=GJcJE1urd1vS8CEi-0)
 - confirm that the date and time matches your selection on the previous page
 - reload the page and observe that it fetches both the referral and provider from API requests, and also that the selection was restored from session
 - click edit and verify that it takes you back to the schedule page
